src/databricks/labs/ucx/source_code/graph.py

```diff
@@ -199,6 +199,46 @@ def visit(self, visit_node: Callable[[DependencyGraph], bool | None], visited: s
     def new_dependency_graph_context(self):
         return DependencyGraphContext(parent=self, path_lookup=self._path_lookup, session_state=self._session_state)
 
+    def _compute_route(self, root: Path, leaf: Path, visited: set[Path]) -> list[Dependency]:
+        # leaf is not on the route so handle case where it doesn't inherit context
+        maybe = self.locate_dependency(leaf)
+        assert maybe.graph  # assume caller knows what they're doing
```

Comment:
don't assume things like that. explicitly return a dependency problem if this occurs.

Reply:
done